aspnetcore/release-notes/aspnetcore-3.1.md

```diff
@@ -0,0 +1,87 @@
+---
+title: What's new in ASP.NET Core 3.1
+author: rick-anderson
+description: Learn about the new features in ASP.NET Core 3.1.
+ms.author: riande
+ms.custom: mvc
+ms.date: 11/12/2019
+no-loc: [Blazor, SignalR]
+uid: aspnetcore-3.1
+---
+# What's new in ASP.NET Core 3.1
+
+This article highlights the most significant changes in ASP.NET Core 3.1 with links to relevant documentation.
+
+## Partial class support for Razor components
+
+Razor components are now generated as partial classes. Code for a Razor component can be written using a code-behind file defined as a partial class rather than defining all the code for the component in a single file. For more information, see [Partial class support](xref:blazor/components#partial-class-support).
+
+## Blazor Component Tag Helper and pass parameters to top-level components
+
+In Blazor with ASP.NET Core 3.0, components were rendered into pages and views using an HTML Helper (`Html.RenderComponentAsync`). In ASP.NET Core 3.1, render a component from a page or view with the new Component Tag Helper:
+
+```razor
+<component type="typeof(Counter)" render-mode="ServerPrerendered" />
+```
+
+The HTML Helper remains supported in ASP.NET Core 3.1, but the Component Tag Helper is recommended.
+
+Blazor Server apps can now pass parameters to top-level components during the initial render. Previously you could only pass parameters to a top-level component with [RenderMode.Static](xref:Microsoft.AspNetCore.Mvc.Rendering.RenderMode.Static). With this release, both [RenderMode.Server](xref:Microsoft.AspNetCore.Mvc.Rendering.RenderMode.Server) and [RenderModel.ServerPrerendered](xref:Microsoft.AspNetCore.Mvc.Rendering.RenderMode.ServerPrerendered) are supported. Any specified parameter values are serialized as JSON and included in the initial response.
+
+For example, prerender a `Counter` component with an increment amount (`IncrementAmount`):
+
+```razor
+<component type="typeof(Counter)" render-mode="ServerPrerendered" 
+    param-IncrementAmount="10" />
+```
+
+For more information, see [Integrate components into Razor Pages and MVC apps](xref:blazor/components#integrate-components-into-razor-pages-and-mvc-apps).
+
+## Support for shared queues in HTTP.sys
+
+[HTTP.sys](xref:fundamentals/servers/httpsys) supports creating anonymous request queues. In ASP.NET Core 3.1, we’ve added to ability to create or attach to an existing named HTTP.sys request queue. Creating or attaching to an existing named HTTP.sys request queue enables scenarios where the HTTP.Sys controller process that owns the queue is independent of the listener process. This independence makes it possible to preserve existing connections and enqueued requests between listener process restarts:
+
+[!code-csharp[](sample/Program.cs?name=snippet)]
+
+<!-- TODO
+## Breaking changes for SameSite cookies
```

Comment:
The behavior of SameSite cookies has changed to reflect upcoming browser changes.  This may affect authentication scenarios like AzureAd, OpenIdConnect, or WsFederation. See (link to new SameSite doc).